presto-native-tests/src/test/java/com.facebook.presto.nativetests/AbstractTestQueriesNative.java

```diff
@@ -0,0 +1,1208 @@
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.testing.MaterializedResult.resultBuilder;
+import static com.facebook.presto.testing.assertions.Assert.assertEquals;
+import static com.facebook.presto.tests.QueryAssertions.assertEqualsIgnoreOrder;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static org.testng.Assert.assertNotEquals;
+
+public abstract class AbstractTestQueriesNative
+        extends AbstractTestQueries
+{
+    private static final String UNSUPPORTED_CORRELATED_SUBQUERY_ERROR_MSG = "line .*: Given correlated subquery is not supported";
+    private static final String P4_HLL_TIMESTAMP_TZ_TYPE_UNSUPPORTED_ERROR = ".*Failed to parse type \\[P4HyperLogLog]. Type not registered.*";
+    private static final String KHLL_TIMESTAMP_TZ_TYPE_UNSUPPORTED_ERROR = ".*inferredType Failed to parse type \\[KHyperLogLog]. Type not registered.*";
+    private static final String CHAR_TYPE_UNSUPPORTED_ERROR = ".*Failed to parse type \\[char\\(.*\\)]. syntax error, unexpected LPAREN, expecting WORD.*";
+    private static final String HASH_GENERATION_UNSUPPORTED_ERROR = ".*Scalar function name not registered: presto.default.\\$operator\\$.*hash.*";
+    private static final String ARRAY_COMPARISON_UNSUPPORTED_ERROR = ".*ARRAY comparison not supported for values that contain nulls.*";
+    private static final String CREATE_HLL_FUNCTION_NOT_REGISTERED = ".*Scalar function name not registered: presto.default.create_hll, called with arguments.*";
+
+    @DataProvider(name = "use_default_literal_coalesce")
+    public static Object[][] useDefaultLiteralCoalesce()
+    {
+        return new Object[][] {{true}};
+    }
+
+    // These tests are disabled because they use the following functions that will not be supported in Presto C++:
+    // custom_add, custom_sum, custom_rank, apply.
+    @Override
+    @Test(enabled = false)
+    public void testCustomAdd() {}
+
+    @Override
+    @Test(enabled = false)
+    public void testCustomSum() {}
+
+    @Override
+    @Test(enabled = false)
+    public void testCustomRank() {}
+
+    @Override
+    @Test(enabled = false)
+    public void testApplyLambdaRepeated() {}
+
+    @Override
+    @Test(enabled = false)
+    public void testLambdaCapture() {}
+
+    @Override
+    @Test(enabled = false)
+    public void testLambdaInAggregationContext() {}
+
+    @Override
+    @Test(enabled = false)
+    public void testLambdaInSubqueryContext() {}
+
+    @Override
+    @Test(enabled = false)
+    public void testNonDeterministicInLambda() {}
+
+    @Override
+    @Test(enabled = false)
+    public void testRowSubscriptInLambda() {}
+
+    @Override
+    @Test(enabled = false)
+    public void testTryWithLambda() {}
+
+    // The following tests are disabled because Velox does not support optimized hash generation.
+    @Override
+    @Test(enabled = false)
```

Comment:
Should we disable all these tests or always force optimizeHashGeneration to false ? That might be better.